handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+
+    **Multi node** refers to configurations where the file store or databases used by SystemLink are running on distinct hardware from the SystemLink application server. This is the recommended configuration for all production deployments.
+
+### NI-SystemLink-Migration Command Line Utility
+
+The workflows for upgrades and migrations make use of the SystemLink command line migration tool, `nislmigrate`. Refer to the readme at the [NI-SystemLink-Migration](https://github.com/ni/NI-SystemLink-Migration) GitHub repo for details on installing and using this tool.
+
+!!!note "Argument Flags for `nislmigrate`"
+    `nislmigrate` supports capturing data from individual services as well as all installed services using the `--all` argument flag. For brevity `--all` is used most workflows in this chapter. Depending on your needs you may replace the `--all` argument flag with one or more of the individual service argument flags.
+
+## Recommended Upgrade and Migrations Workflows for your deployment
+
+This chapter recommends workflows for the following upgrade and migration scenarios. You should plan for some downtime of your SystemLink Server. By following these recommendations this downtime can be minimized.
+
+- [Single Node Upgrade](#single-node-upgrade)
+- [Single Node to Multi Node Migration](#single-node-to-multi-node-migration)
+    - [Single Node to Multi Node with MongoDB](#single-node-to-multi-node-with-mongodb)
+    - [Single Node to Multi Node with File Storage](#single-node-to-multi-node-with-file-storage)
+    - [Single Node to Multi Node with PostGreSQL](#single-node-to-multi-node-with-postgresql)
+- [Upgrading Multi node configurations (MongoDB, PostGreSQL, File Share/S3)](#upgrading-multi-node-configurations)
+- [Seamless cut-over](#seamless-cut-over)
+
+!!!note "Storing Captured Data"
+    For upgrades and migrations for a SystemLink Server with a small amount of data you may be able to use `nislmigrate` to backup your data to the local file system of your SystemLink Server. For SystemLink Servers with a large amount of data be aware `nislmigrate` could exhaust the local storage on the server. This can mitigated by using an attached volume. **This document assumes this will be needed and refers to this attached volume as `D:\` in each workflow.**
+
+## Single Node Upgrade
+
+This workflow describes steps to upgrade a single node deployment of SystemLink Server.
+
+The SystemLink NI Package Manager Installers (NIPM) supports in-place upgrades of SystemLink Server. An in place upgrade is one where the upgrade is run directly on your current SystemLink Server. For production deployments of SystemLink Server this is not recommended. If you pursue this option it is highly recommended you backup your server prior to the upgrade. This backup is an essential fallback should an issue occur during upgrade that leaves your SystemLink Server in a bad state.
```

Comment:
```suggestion
The NI Package Manager (NIPM) installer for SystemLink supports in-place upgrades of SystemLink Server. An in-place upgrade is one where the upgrade is run directly on your current SystemLink Server. NI does not recommend in-place upgrades for production deployments of SystemLink Server. If you pursue this option it is highly recommended that you backup your server prior to the upgrade. This backup is an essential fallback should an issue occur during upgrade that leaves your SystemLink Server in a bad state.
```

Reply:
Though the NI Package Manager installer for SystemLink supports in-place upgrades where the upgrade runs directly on your current SystemLink Server, NI does not recommend this option. If you choose this option, ensure that you backup your server before beginning the upgrade.